### PR TITLE
fix(artifact): ignore auth header when downloading external artifacts

### PR DESCRIPTION
**Pull request by ps610**

### CODE REVIEW

File: artifacts/ArtifactHandling/Invoke-DownloadArtifact.ps1 [hunk 1]
```diff
@@ -191,9 +191,12 @@ function Invoke-DownloadArtifact {
                 if ("$sourceUri".StartsWith("http")) { 
                     # ignore auth header if SAS token is present
                     if ($sourceUri -match "([?&](sp|st|se|sv|sr|sig)=[^&]+)") {
-                        $headers = @{}
+                        Add-ArtifactsLog -message "SAS token found in URL, ignoring Authorization header" -severity Debug
```
Comment (lippertmarkus) [inline artifacts/ArtifactHandling/Invoke-DownloadArtifact.ps1]: to be generic, wouldn't it be enough to check if the url starts with the PublicDnsName and if not, to not send the auth header?

Comment (ps610) [reply artifacts/ArtifactHandling/Invoke-DownloadArtifact.ps1]: in general yes, but there might be old containers where the env variable is not set yet (as we implemented this at the end of August)

Comment (ps610) [reply artifacts/ArtifactHandling/Invoke-DownloadArtifact.ps1]: Which is also the reason for the fallback logic here:
https://github.com/cosmoconsult/container-scripts-for-selfservice/blob/49a34ab1ab015f99478b82fb874882548d478da3/artifacts/ArtifactHandling/Invoke-DownloadArtifact.ps1#L90-L92

Comment (ps610) [reply artifacts/ArtifactHandling/Invoke-DownloadArtifact.ps1]: ah nvm, I'll improve it

**Review by lippertmarkus — APPROVED** [review]

**Pull request merged**